ProjectBank/Core/IProjectRepository.cs

```diff
@@ -1,5 +1,12 @@
 namespace ProjectBank.Core;
-
-public class IProjectRepository
+public interface IProjectRepository
 {
+    Task<ProjectDTO> CreateProjectAsync(ProjectCreateDTO project);
+    Task<IReadOnlyCollection<ProjectDTO>> ReadAllAsync();
+    Task<ProjectDTO> ReadProjectAsync(int projectId);
```

Comment:
This should be renamed to "ReadProjectByIdAsync" to be in line with the other methods that takes a specific id.